### PR TITLE
Feature/add dataset template

### PR DESCRIPTION
**Pull request by JoshSinghONS**
### What

Added dataset page model for template migration.

### How to review

Sanity check.

### Who can review

Anyone? I think.

### CODE REVIEW

File: model/datasetPage/model.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package datasetPage
+
+import "github.com/ONSdigital/dp-frontend-models/model"
+
+//Page contains data re-used for each page type a Data struct for data specific to the page type
+type Page struct {
+	model.Page
+	DatasetPage DatasetPage `json:"data"`
```
Comment (JoshSinghONS) [inline model/datasetPage/model.go:8]: @mr-nick17 

Having trouble understanding what this line means.

File: model/datasetPage/model.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+//Related content (split by type) to this page
+type Related struct {
+	Publications       []model.Related `json:"related_publications"`
+	FilterableDatasets []model.Related `json:"related_filterable_datasets"`
+	Datasets           []model.Related `json:"related_datasets"`
+	Methodology        []model.Related `json:"related_methodology"`
+	Links              []model.Related `json:"related_links"`
+}
```
Comment (JoshSinghONS) [inline model/datasetPage/model.go]: @mr-nick17 struggling to see where this struct is used?

File: model/datasetPage/model.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+//SupplementaryFile is a downloadable file that is associated to an individual dataset
+type SupplementaryFile struct {
+	Title     string `json:"title"`
+	Extension string `json:"extension"`
+	Size      string `json:"size"`
+	URI       string `json:"uri"`
+}
+
+type Version struct {
```
Comment (daiLlew) [inline model/datasetPage/model.go:43]: Godoc missing on this one.

File: model/datasetPage/model.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	model.ContactDetails
+}
+
+//Dataset has the file and title information for an individual dataset
+type DatasetPage struct {
+	Versions            []Version           `json:"versions"`
+	SupplementaryFiles  []SupplementaryFile `json:"supplementary_files"`
+	Downloads           []Download          `json:"downloads"`
+	IsNationalStatistic bool                `json:"national_statistic"`
+	ReleaseDate         string              `json:"releaseDate"`
```
Comment (daiLlew) [inline model/datasetPage/model.go]: I think our json naming convention requires these to be `release_date`, `next_release`, `dataset_id`etc.

**Review by daiLlew — COMMENTED** [review]
Minor comments but otherwise LGTM.

File: model/datasetPage/model.go (hunk 4 above)
Comment (nshumoogum) [reply model/datasetPage/model.go]: `snake_case`

File: model/datasetPage/model.go [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	Extension string `json:"extension"`
+	Size      string `json:"size"`
+	URI       string `json:"uri"`
+}
+
+//Version has the details for a previous version of the dataset
+type Version struct {
+	URI              string     `json:"url"`
+	UpdateDate       string     `json:"updateDate"`
+	CorrectionNotice string     `json:"correctionNotice"`
```
Comment (daiLlew) [inline model/datasetPage/model.go]: Same here (Sorry missed these before).

**Review by daiLlew — APPROVED** [review]

**Pull request merged**